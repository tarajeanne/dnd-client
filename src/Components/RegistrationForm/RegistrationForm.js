import React, { Component } from 'react';
import { Button, Input, Required } from '../../Utils/Utils';
import AuthApiService from '../../services/auth-api-service';

class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  };

  state = { error: null };

  handleSubmit = (ev) => {
    ev.preventDefault();
    const { username, password } = ev.target;

    this.setState({ error: null });
    AuthApiService.postUser({
      username: username.value,
      password: password.value,
    })
      .then((user) => {
        username.value = '';
        password.value = '';
        this.props.onRegistrationSuccess(user);
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;
    return (
      <form className="RegistrationForm" onSubmit={this.handleSubmit}>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <div className="username">
          <label htmlFor="RegistrationForm__username">
            Username:  <Required />
          </label>
          <Input
            name="username"
            type="text"
            required
            id="RegistrationForm__username"
          />
        </div>
        <div className="password">
          <label htmlFor="RegistrationForm__password">
            Password:  <Required />
          </label>
          <Input
            name="password"
            type="password"
            required
            id="RegistrationForm__password"
          />
        </div>
        <Button type="submit">Register</Button>
      </form>
    );
  }
}

export default RegistrationForm;
