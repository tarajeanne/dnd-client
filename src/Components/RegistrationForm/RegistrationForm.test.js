import React from 'react';
import RegistrationForm from './RegistrationForm';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('RegistrationForm', () => {
    it('renders as expected', () => {
        const wrapper = shallow(<RegistrationForm />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
  });
