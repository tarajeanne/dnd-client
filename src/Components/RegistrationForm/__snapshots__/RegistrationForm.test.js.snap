// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm renders as expected 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="username"
  >
    <label
      htmlFor="RegistrationForm__username"
    >
      Username:  
      <Required />
    </label>
    <Input
      id="RegistrationForm__username"
      name="username"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password:  
      <Required />
    </label>
    <Input
      id="RegistrationForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <Button
    type="submit"
  >
    Register
  </Button>
</form>
`;
